### PR TITLE
Remove api-server --insecure-port for versions greater than 1.20

### PR DESCRIPTION
**Pull request by jsturtevant**
**Reason for Change**:

This flag was removed in 1.24+  https://github.com/kubernetes/kubernetes/commit/9573b4a6b993aaa2238d347419ffe7c4b8005c94 and windows jobs are failing: https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-aks-engine-azure-master-windows-containerd/1469025482350530560

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [ ] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [x] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/api/defaults-apiserver.go [hunk 1]
```diff
@@ -105,6 +104,10 @@ func (cs *ContainerService) setAPIServerConfig() {
 		defaultAPIServerConfig["--service-account-signing-key-file"] = "/etc/kubernetes/certs/apiserver.key"
 	}
 
+	if !common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.24.0-alpha") {
```
Comment (jackfrancis) [inline pkg/api/defaults-apiserver.go:107]: Let's change this to `1.24.0-alpha.0`

Comment (jackfrancis) [reply pkg/api/defaults-apiserver.go:107]: Also cc @alexeldeib in case AKS needs to update its kubelet configuration surface area.

Comment (jsturtevant) [reply pkg/api/defaults-apiserver.go]: as discussed moved this to < 1.20

**Comment by jsturtevant** [comment]
The flags are defaulted for 1.20+ so we are removing it for all version gt 1.20.  1.19 is out of support upstream but currently still in aks-e

https://github.com/kubernetes/kubernetes/commit/cfc2b330a7027e109ac26fc6a86b68898883099e

**Comment by jsturtevant** [comment]
deployment working with main branch of k8s:

```
 kubectl --kubeconfig ~/out/test-upstream-artifacts-9091/kubeconfig/kubeconfig.westus2.json get nodes                                                                                                                                                                                                       NAME                    STATUS   ROLES    AGE     VERSION 
1072k8s000              Ready    agent    6m29s   v1.24.0-alpha.1.55+1b0d83f1d63b4e
1072k8s001              Ready    agent    6m2s    v1.24.0-alpha.1.55+1b0d83f1d63b4e
k8s-master-10728554-0   Ready    master   7m24s   v1.24.0-alpha.1.55+1b0d83f1d63b4e 
```

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Pull request merged**